modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_selector/AssetCategoriesSelector.es.js

```diff
@@ -12,64 +12,77 @@
  * details.
  */
 
-import 'clay-icon';
-import Component from 'metal-component';
-import Soy from 'metal-soy';
-import {Config} from 'metal-state';
+import PropTypes from 'prop-types';
+import React from 'react';
 
-import './AssetVocabularyCategoriesSelector.es';
-import templates from './AssetCategoriesSelector.soy';
+import AssetVocabularyCategoriesSelector from './AssetVocabularyCategoriesSelector.es';
 
-/**
- * AssetCategoriesSelector is a component wrapping the existing Clay's MultiSelect component
- * that offers the user a tag selection input
- */
-class AssetCategoriesSelector extends Component {
-	_handleInputFocus(event) {
-		this.emit('inputFocus', event);
-	}
-
-	_handleSelectedItemsChange(event) {
-		this.emit('selectedItemsChange', event);
-	}
-}
+function AssetCategoriesSelector({
+	eventName,
+	groupIds,
+	id,
+	inputName,
+	onVocabulariesChange = () => {},
```

Comment:
This (providing a default no-op function) is a pattern I see a bit and I think we might want to consider avoiding it.

In some cases (and I think this is probably one of them), the component is useless if you don't pass it a function, so we should make it required and not provide a default.

In other cases (not here, but in other places I've seen this), the default gets used a lot and you end up recreating this no-op function on every render; if the function ends up getting passed around to other places (eg. via props to other functions, or to hooks) then the fact that it is not stable over time can cause re-renders, even though it is actually "the same" (or at least, is totally equivalent) function.

In both cases, the "fix" is simple enough. Just make sure `onVocabulariesChange` exists before calling it.

Reply:
>  In some cases (and I think this is probably one of them), the component is useless if you don't pass it a function, so we should make it required and not provide a default.

Yes that's totally the case here. Going to mark this prop as required.